### PR TITLE
feat!: add support for custom_placement_config

### PR DESCRIPTION
**Pull request by johnksv**
Add support for custom_placement_config which was added to provider in version [4.40.0](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4400-october-10-2022).

`custom_placement_config` is used for dual-region bucket configuration.

Docs for custom_placement_config: https://registry.terraform.io/providers/hashicorp/google/4.56.0/docs/resources/storage_bucket#custom_placement_config

### CODE REVIEW

**Comment by johnksv** [comment]
Google Individual Contributor License Agreement signed.

**Comment by apeabody** [comment]
Hi @johnksv - Thanks for the contribution, some relevant INT output:

```
Error: googleapi: Error 400: Project - may not create storageClass MULTI_REGIONAL buckets with locationConstraint EU and data_location europe-north1, data_location europe-west1.., invalid
```

**Comment by johnksv** [comment]
Hi @apeabody 
Thanks.
I updated the example in simple bucket instead of multiple_buckets, due to the lifecycle in multiple_buckets causing collision with dual region/custom_placement_config setup.

**Comment by johnksv** [comment]
Merged main into this branch.
Please let me know if you would like it to be a rebase instead.

**Comment by johnksv** [comment]
Hi again, @apeabody 
Anything missing from this getting merged? Thanks!

File: modules/simple_bucket/variables.tf [hunk 1]
```diff
@@ -78,6 +78,14 @@ variable "retention_policy" {
   default = null
 }
 
+variable "custom_placement_config" {
+  description = "Configuration of the bucket's custom location in a dual-region bucket setup. If the bucket is designated a single or multi-region, the variable are null."
+  type = object({
```
Comment (apeabody) [inline modules/simple_bucket/variables.tf:83]: Is this syntax supported in TF 0.13?  

https://github.com/terraform-google-modules/terraform-google-cloud-storage/blob/master/modules/simple_bucket/versions.tf#L18

**Review by apeabody — COMMENTED** [review]
Thanks for the contribution @johnksv!  Quick Questions on one section.

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @johnksv!

**Pull request merged**